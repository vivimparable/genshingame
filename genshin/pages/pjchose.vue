<template>
    <div>
        <h2>Wich character are you looking for</h2>
        <p>NOTE: Characters with 2 names have to be written with a - split.</p>
        <h6>Example: genshin-impact</h6>


        <label for="Character">Character name:</label>
        <input v-if="!this.champExists" v-model="charactername" type="text" name="" id="">
        
        <p>There's <span :class="[{'text-danger': arryPj > 0},{'text-success' : arryPj == 0}]" >{{arryPj-3}}</span> champions left.</p>
            <hr>
        <div class="row">

        <div   v-if="!champExists" class="  col-3 ">
            <h4>Champions found</h4>
             <ul v-for="i of championsGottren" :key="i">
            <li>{{i}}</li>
        </ul>
        
        </div >

        <div  class=" p-3 " v-else>
            <h3 class="text-light p-5 bg-danger">This champ is already on.</h3>
            <button class="btn btn-danger " @click="champExists = false">Back</button>
        </div>
       
        <div class="col-9 border-top border-left border-right">
            <h4>Info of the champion found</h4>
            <nuxt-child v-on:Championgotten="GottenChamp" />

        </div>
    </div>
    </div>
</template>

<script>
import axios from 'axios'

export default {
    
    name:'pjchose',
    data(){
        return{
            charactername:'',
            championsGottren:[],
            arryPj:null,
            champExists:false
        }
    },watch:{
        
        charactername(a){
            if(a){
                this.$router.push({path: '/pjchose/'+ this.charactername })
            }
        }
    },methods:{
        GottenChamp(a){
            const booleanFindChamp = this.championsGottren.find(champ => champ === a)

            if(booleanFindChamp){

                this.champExists=true
                

            }else{
                this.championsGottren.push(a)
                this.arryPj--;
            }
            
        }
    },async created(){

            try {
                const resp = await axios.get('https://api.genshin.dev/characters')
                this.arryPj= resp.data.length
                
            } catch (error) {
                
            }
    }
}
</script>

<style  scoped>
ul{
    list-style-type: none;
}

span{
    font-size: 2rem;
}
</style>

