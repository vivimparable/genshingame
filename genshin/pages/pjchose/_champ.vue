<template>
<div>
    <div  v-if="this.champion.length==0">
        <p>No hay un personaje asi</p>
    </div>
    <div class="description p-3" v-else >  
        
        <h1>{{champion.name}}</h1>
        <hr>
        <h3>Element: {{champion.vision}}</h3>
        <h3>Weapon: {{champion.weapon}}</h3>
        <h3>From: {{champion.nation}}</h3>
        <h3>Stars: {{champion.rarity}}/5</h3>
        <h3>Birthday: {{champion.birthday}}</h3>
        <h3>Description:</h3><p>{{champion.description}}</p>

            <hr>
        <h3>Constellations</h3> 
        <div v-for="(i,index) of this.champion.constellations" :key="index">

            <p>{{index+1}}:
                <span>  {{i.name}}</span>
               
                . {{i.description}}
              </p>

        </div>
    </div>  
</div>    
</template>

<script>
import axios from 'axios'
export default {
    data(){
        return{
            champName:this.$route.params.champ,
            champion:[],

        }
    },
  
  async created(){
    try {
      const response = await axios.get('https://api.genshin.dev/characters/'+ this.$route.params.champ)
      this.champion=response.data
        this.pushElement()
    } catch (error) {
    }
  },watch:{
    async champName(){
    try {
      const response = await axios.get('https://api.genshin.dev/characters/'+ this.$route.params.champ)
      this.champion=response.data
      this.pushElement()

      console.log(this.champion)
    } catch (error) {
     
    }
    }
  },methods:{
    pushElement(){
      if(this.champion){
        const championEmitted = this.champion.name;
        this.$emit('Championgotten',championEmitted )
      }
    }
  }
}
</script>

<style scoped>
span{
 font-weight: bold;
}

.description{
  background-color: rgb(204 204 204 );
}
</style>