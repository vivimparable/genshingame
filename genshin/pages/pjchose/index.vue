<template>
    <div>
        <p>Start typing</p>
    </div>
</template>
