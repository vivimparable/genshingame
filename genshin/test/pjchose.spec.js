import { shallowMount } from '@vue/test-utils'
import pjchose from '@/pages/pjchose.vue'

describe('MainPageGenshim', () => {
 let wrapper = shallowMount(pjchose)
 beforeEach(()=>{
  wrapper = shallowMount(pjchose)

})
      test('Snapshot', () => {
        expect(wrapper.element).toMatchSnapshot()
      })

      test('Element', () => {
        const wrapper = shallowMount(pjchose,{
          data(){
            return{
                charactername:'Yelan'
            }
          }

        })

        let characterName = wrapper.vm.$options.data().charactername;

        expect(characterName).toBe('Yelan')

        let funcionName =  wrapper.vm.$options.watch.charactername;
        
        console.log(wrapper.vm.$options)
        
      })

})
