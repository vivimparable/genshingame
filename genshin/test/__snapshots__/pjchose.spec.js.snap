// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPageGenshim Snapshot 1`] = `
<div>
  <h2>
    Wich character are you looking for
  </h2>
   
  <label
    for="Character"
  >
    Character name:
  </label>
   
  <input
    id=""
    name=""
    type="text"
  />
   
  <nuxt-child />
</div>
`;
